sunpy/net/dataretriever/sources/vsm.py

```diff
@@ -0,0 +1,163 @@
+"""
+This module implements SOLIS VSM.
+"""
+#This module was developed with funding provided by
+#the Google Summer of Code 2016.
+
+__author__ = "Sudarshan Konge"
+__email__ = "sudk1896@gmail.com"
+
+import datetime
+import urllib2
+import re
+from bs4 import BeautifulSoup
+import numpy as np
+
+from sunpy.net.dataretriever.client import GenericClient
+from sunpy.util.scraper import Scraper
+
+_all__ = ['VSMClient']
+
+class VSMClient(GenericClient):
+    """
+    Returns a list of URLS to SOLIS VSM files corresponding to value of input timerange.
+    URL source: `http://gong2.nso.edu/pubkeep/`.
+    Parameters
+    ----------
+    timerange: sunpy.time.TimeRange
+        time range for which data is to be downloaded.
+        Example value - TimeRange('2015-12-30 00:00:00','2015-12-31 00:01:00')
+
+    Instrument: Fixed argument - 'vsm' ,case insensitive.
+
+    Wavelength: Any of 6302, 8542 or 10830 Angstrom units or equivalent values in different units.
+                Expects a Wavelength object.
+                e.g. 6302*u.AA, 630.2*u.nm etc.
+
+    Physobs: Optional, includes 'LOS_MAGNETIC_FIELD', "VECTOR_MAGNETIC_FIELD'
+             'EQUIVALENT_WIDTH'.
+
+    Examples
+    --------
+    >>> from sunpy.net import Fido
+    >>> from sunpy.net import attrs as a
+    >>> res = Fido.search(a.Time('2015/1/3', '2015/1/6'), a.Instrument('vsm'), a.Wavelength(630.2*u.nm),
+                     a.Physobs('VECTOR_MAGNETIC_FIELD'))
+    >>> print (res)
+    [<Table length=6>
+         Start Time           End Time      Source Instrument
+           str19               str19         str5     str3
+    ------------------- ------------------- ------ ----------
+    2015-01-03 00:00:00 2015-01-04 00:00:00  SOLIS        vsm
+    2015-01-04 00:00:00 2015-01-05 00:00:00  SOLIS        vsm
+    2015-01-05 00:00:00 2015-01-06 00:00:00  SOLIS        vsm
+    2015-01-06 00:00:00 2015-01-07 00:00:00  SOLIS        vsm
+    2015-01-07 00:00:00 2015-01-08 00:00:00  SOLIS        vsm
+    2015-01-08 00:00:00 2015-01-09 00:00:00  SOLIS        vsm]
+    """
+    def _get_url_for_timerange(self, timerange, **kwargs):
+        """
+        returns list of urls corresponding to given TimeRange.
+        """
+        table_physobs = ['EQUIVALENT_WIDTH', 'LOS_MAGNETIC_FIELD', 'VECTOR_MAGNETIC_FIELD']
+        table_wave = {6302:'v72', 8542:'v82', 10830:'v22'}
+
+        physobs_in = ('physobs' in kwargs.keys() and kwargs['physobs'] in table_physobs)
+##        url_pattern = 'http://gong2.nso.edu/pubkeep/{dtype}/%Y%m/k4{dtype}%y%m%d/k4{dtype}%y%m%dt%H%M%S{suffix}'
+
+        wave_float = kwargs['wavelength'].min.value
```

Comment:
All values in ```Wavelength``` class, be any unit are converted to Angstroms automatically. It is there in the docs.